### PR TITLE
FADEC power supply logic

### PR DESCRIPTION
**Pull request by tyler58546**
**Summary of Changes**

* Added engine FADEC power supply logic
* FADECs will now be powered for 5 minutes after initial electrical power and 5 minutes after engine shutdown
* FADECs will now be powered when engine mode selector is set to ignition
* FADECs will now be powered when N2 > 15%
* FADECs will now lose power when the engine fire p/b is pressed

**References**

image

### CODE REVIEW

**Review by AdenFlorian — APPROVED** [review]
ready to test!

**Comment by Snapmatics** [comment]
Can we have some more reference if possible?

**Comment by tyler58546** [comment]
> 
> 
> Can we have some more reference if possible?

This PR changes when the values on the engine display show XX.
The values on the engine display show XX when the engines FADEC system is not powered.
Previously, the FADECs would only be powered when the engine is running or when the engine master switch was set to on.
With this PR, the FADECs now use the correct power supply logic shown above.

**Comment by Snapmatics** [comment]
> > Can we have some more reference if possible?
> 
> This PR changes when the values on the engine display show XX.
> The values on the engine display show XX when the engines FADEC system is not powered.
> Previously, the FADECs would only be powered when the engine is running or when the engine master switch was set to on.
> With this PR, the FADECs now use the correct power supply logic shown above.

Got it, thx.

File: A32NX/html_ui/Pages/A32NX_Core/A32NX_FADEC.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    updateSimVars() {
+        this.lastDCState = SimVar.GetSimVarValue("L:DCPowerAvailable", "Bool");
+        this.lastMasterState = SimVar.GetSimVarValue("FUELSYSTEM VALVE SWITCH:"+(this.engine+5), "Bool");
+        SimVar.SetSimVarValue("L:A32NX_FADEC_POWERED_ENG"+this.engine, "Bool", this.isPowered() ? 1 : 0);
+    }
+
+    isPowered() {
+        if (SimVar.GetSimVarValue("L:FIRE_BUTTON_ENG"+this.engine, "Bool") == 1) return false;
+        if (SimVar.GetSimVarValue("TURB ENG N2:"+this.engine, "Percent") > 15) return true;
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/A32NX_Core/A32NX_FADEC.js:32]:
```suggestion
        if (SimVar.GetSimVarValue("L:FIRE_BUTTON_ENG" + this.engine, "Bool") == 1) return false;
        if (SimVar.GetSimVarValue("TURB ENG N2:" + this.engine, "Percent") > 15) return true;
```

**Comment by tareksabet** [comment]
everything works well

**Comment by FlyingCarpet7** [comment]
Great implementation!
Just a comment about the note:
 *If ENG MODE selector is set to NORM position before ENG start the FADEC supply is cut if. *
This is correct but it has a timer. I timed it IRL to 30 seconds before the supply is cut.
If you use the FADEC GND power switches and turn them of, the power supply is cut after only 2-3 seconds.

**Comment by NathanInnes** [comment]
Please include CHANGELOG

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 2]
```diff
@@ -1349,7 +1349,7 @@ var A320_Neo_UpperECAM;
         }
         update(_deltaTime) {
             if (this.allGauges != null) {
-                var active = A320_Neo_ECAM_Common.isEngineDisplayActive(this.index + 1);
+                const active = SimVar.GetSimVarValue("L:A32NX_FADEC_POWERED_ENG"+(this.index+1), "Bool") == 1;
```
Comment (NathanInnes) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1352]:
```suggestion
                const active = SimVar.GetSimVarValue("L:A32NX_FADEC_POWERED_ENG"+(this.index + 1), "Bool") == 1;
```

**Comment by tyler58546** [comment]
PR is compatible with the 1.9.3 update.

**Comment by tyler58546** [comment]
I have added the 30 second timer @FlyingCarpet7 mentioned when the engine mode selector is changed to NORM before engine start.

File: A32NX/html_ui/Pages/A32NX_Core/A32NX_FADEC.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    update(_deltaTime) {
+        const dcAvail = SimVar.GetSimVarValue("L:DCPowerAvailable", "Bool");
+        const masterState = SimVar.GetSimVarValue("FUELSYSTEM VALVE SWITCH:"+(this.engine+5), "Bool");
+        const ignitionState = SimVar.GetSimVarValue("L:XMLVAR_ENG_MODE_SEL", "Enum") == 2;
+        if (this.lastDCState != dcAvail && dcAvail == 1) {
+            this.fadecTimer = 5*60;
+        }
+        if (this.lastMasterState != masterState) {
```
Comment (Meqolo) [inline A32NX/html_ui/Pages/A32NX_Core/A32NX_FADEC.js:18]: Could condense this by replacing line 15 with:
```
 if ((this.lastDCState != dcAvail && dcAvail == 1) || this.lastMasterState != masterState) {
```

**Comment by starvingjavi** [comment]
Could it be possible to add some instructions on how to test this functionality?

File: A32NX/html_ui/Pages/A32NX_Core/A32NX_FADEC.js (hunk 3 above)
Comment (wpine215) [reply A32NX/html_ui/Pages/A32NX_Core/A32NX_FADEC.js]: Implemented.

**Pull request merged**